examples/model_compress/model_prune_torch.py

```diff
@@ -145,7 +145,10 @@ def create_model(model_name='naive'):
 def create_pruner(model, pruner_name, optimizer=None):
     pruner_class = prune_config[pruner_name]['pruner_class']
     config_list = prune_config[pruner_name]['config_list']
-    return pruner_class(model, config_list, optimizer)
+    if pruner_name in ['level', 'slim', 'fpgm', 'l1filter']:
```

Comment:
why keep optimizer parameter for `ActivationMeanRankFilterPruner` and `ActivationAPoZRankFilterPruner`? they do not need optimizer hook either.  For all one-shot pruners, it seems only TaylorFOWeightFilterPruner needs optimizer ( to calculate gradient contribution).

Reply:
> why keep optimizer parameter for `ActivationMeanRankFilterPruner` and `ActivationAPoZRankFilterPruner`? they do not need optimizer hook either. For all one-shot pruners, it seems only TaylorFOWeightFilterPruner needs optimizer ( to calculate gradient contribution).

In the current implementation, `ActivationMeanRankFilterPruner` and `ActivationAPoZRankFilterPruner` need the optimizer hook.

Reply:
as discussed in the meeting, let's not do this broken change?